ngx_http_access_token_to_jwt.c

```diff
@@ -414,8 +436,41 @@ static char *ngx_http_access_token_to_jwt_merge_loc_conf(ngx_conf_t *config, voi
     ngx_http_access_token_to_jwt_conf_t *prev = parent, *conf = child;
 
     ngx_conf_merge_value(conf->enable, prev->enable, 0);
-    ngx_conf_merge_str_value(conf->base64encoded_client_credentials, prev->base64encoded_client_credentials, "");
+
+    ngx_conf_merge_str_value(conf->client_id, prev->client_id, "");
+    ngx_conf_merge_str_value(conf->client_secret, prev->client_secret, "");
     ngx_conf_merge_str_value(conf->introspection_endpoint, prev->introspection_endpoint, "");
 
+
+    ngx_str_t  *concat_credentials =  ngx_pcalloc(config->pool, sizeof(ngx_str_t));
+
+    if (concat_credentials == NULL)
+    {
+        return NGX_CONF_ERROR;
+    }
+
+    int concat_credentials_size = conf->client_id.len + conf->client_secret.len + 1; // client_id:client_secret
```

Comment:
I tested it. You were getting lucky is all. You can see this if you add `concat_credentials[concat_credentials_size] = 1;` right after you alloc the memory for `concat_credentials`. In this case, you'll always have an ASCII Start of Header (SOH) char at the end of your password. This is fixed in 91c44ac where I added a macro, `basic_credential_length` to calculate the length of the header like this:

```c
#define basic_credential_length(id, secret) id + (sizeof(":") - 1) + secret + (sizeof("\0") - 1)
```

With this fixed and the other stuff you did, this branch is ready to be merged once the conflicts are resolved. I'll leave that to you.